rxgroups/src/main/java/com/airbnb/rxgroups/ResubscribeHelper.java

```diff
@@ -9,54 +9,66 @@
 
 class ResubscribeHelper {
 
-    private static final Map<Class<?>, Constructor<?>> BINDINGS = new LinkedHashMap<>();
-
-    static void initializeResubscription(Object target, ObservableGroup group) {
-        Constructor<?> constructor = findConstructorForClass(target.getClass(), group);
-        if (constructor == null) {
-            throw new IllegalArgumentException("Invalid target: " + target);
-        }
-
-        try {
-            constructor.newInstance(target, group);
-        } catch (IllegalAccessException e) {
-            throw new RuntimeException("Unable to invoke " + constructor, e);
-        } catch (InstantiationException e) {
-            throw new RuntimeException("Unable to invoke " + constructor, e);
-        } catch (InvocationTargetException e) {
-            Throwable cause = e.getCause();
-            if (cause instanceof RuntimeException) {
-                throw (RuntimeException) cause;
-            }
-            if (cause instanceof Error) {
-                throw (Error) cause;
-            }
-            throw new RuntimeException("Unable to create resubscribe instance.", cause);
-        }
+  private static final Map<Class<?>, Constructor<?>> BINDINGS = new LinkedHashMap<>();
+
+  static void initializeResubscription(Object target, ObservableGroup group) {
+    Constructor<?> constructor = findConstructorForClass(target.getClass(), group);
+    if (constructor == null) {
+      throw new IllegalArgumentException("Invalid target: " + target);
+    }
+    invokeConstructor(constructor, target, group);
+  }
+
+  static void safeInitializeResubscription(Object target, ObservableGroup group) {
+    Constructor<?> constructor = findConstructorForClass(target.getClass(), group);
+    if (constructor == null) {
+      return;
+    }
+    invokeConstructor(constructor, target, group);
+  }
+
+  private static void invokeConstructor(Constructor<?> constructor, Object target,
+                                        ObservableGroup group) {
+    try {
+      constructor.newInstance(target, group);
+    } catch (IllegalAccessException e) {
+      throw new RuntimeException("Unable to invoke " + constructor, e);
+    } catch (InstantiationException e) {
+      throw new RuntimeException("Unable to invoke " + constructor, e);
+    } catch (InvocationTargetException e) {
+      Throwable cause = e.getCause();
+      if (cause instanceof RuntimeException) {
+        throw (RuntimeException) cause;
+      }
+      if (cause instanceof Error) {
+        throw (Error) cause;
+      }
+      throw new RuntimeException("Unable to create resubscribeAll instance.", cause);
+    }
+  }
+
+  private static Constructor<?> findConstructorForClass(Class<?> cls, ObservableGroup group) {
+    Constructor<?> bindingCtor = BINDINGS.get(cls);
+    if (bindingCtor != null) {
```

Comment:
Good point! I'll add that optimization. 

Now that I think about it, I should probably solve the superclass problem before rolling this into the app. If we want to enable the pattern of a base class calling `onCreate` we will need to subscribe all classes in the hierarchy. Currently, I believe if we have `A extends B extends Base` only `A` would have it's fields auto initialized if Base calls initialize(this).

We could have the initialize code take a class instead of an object, and require each object in the hierarchy to call initialize.

For onCreate, it seems like the proper API would be to initialize the entire super class chain. In the generic case, it may or may not be desired to have the entire super class chain initialized as who is to say that when I call initialize in the base class, I want to call initialize in derived classes?